### DISCUSSION TITLE
[Q&A] Problem with install MSI from GPO or Startup script - glpi-agent version 1.8 and 1.9

### DISCUSSION THREAD

**Discussion by danix1** [glpi-project/glpi-agent#688] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hello, I try to install new version glpi-agent 1.8 and 1.9 on Windows 11 Pro and have problem. I install with GPO > Computer configuration > Software settings > Software installation with MSI file. When I start computer, I only see Waiting, about 20-30 minutes, and GLPI is not installed. Only few files in Program files\Glpi-agent is present. No binary. This install msi file with SYSTEM user.

Same happend, when I use startup/shutdown script. I only change version of msi, and version 1.8 and 1.9 wont install. I try to install with parameters, without, with MST or without MST, still same problem. Also try to remove registry and check if service is not present and installed. Failed install - long waiting to service and ended with error: Service 'GLPI Agent' (glpi-agent) failed to start
I wont use official vbs script, because I don't have internet connection on this PCs.
Installation of older version 1.4 - 1.7.3 working OK.

How to reproduce: Try install MSI with GPO, expected glpi-agent is installed, and service is working.

version 1.5 - install OK
version 1.7.2 - install OK
version 1.7.3 - install OK
version 1.8 - install FAILED
version 1.9 - install FAILED

Please help, thank you.

Is related to? https://github.com/glpi-project/glpi-agent/issues/584

Error from MSI log - waiting about 10 minutes, to start service and service failed to start:

```
MSI (s) (E4:0C) [09:48:42:929]: Executing op: ActionStart(Name=CreateShortcuts,Description=Creating shortcuts,Template=Shortcut: [1])
WixQuietExec:  
MSI (s) (E4:0C) [09:48:42:929]: Executing op: IconCreate(Icon=i_main_ico,Data=BinaryData)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: IconCreate(Icon=agentmonitor.ico,Data=BinaryData)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: ActionStart(Name=WriteRegistryValues,Description=Writing system registry values,Template=Key: [1], Name: [2], Value: [3])
MSI (s) (E4:0C) [09:48:42:929]: Executing op: ProgressTotal(Total=52,Type=1,ByteEquivalent=13200)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegOpenKey(Root=-2147483646,Key=Software\GLPI-Agent,,BinaryType=1,,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=additional-content,,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=debug,Value=0,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=local,,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=logger,Value=file,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=logfile,Value=C:\Program Files\GLPI-Agent\logs\glpi-agent.log,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=logfile-maxsize,Value=4,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=server,Value=https://helpdesk.tsk.sk/plugins/glpiinventory,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=no-httpd,Value=0,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=httpd-ip,Value=0.0.0.0,)
MSI (s) (E4:0C) [09:48:42:929]: Executing op: RegAddValue(Name=httpd-port,Value=62354,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=httpd-trust,Value=127.0.0.1/32,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=tag,,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=scan-homedirs,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=scan-profiles,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=no-p2p,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=timeout,Value=180,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=snmp-retries,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=delaytime,Value=3600,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=backend-collect-timeout,Value=180,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=full-inventory-postpone,Value=14,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=no-task,,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=no-category,,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=no-compression,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=html,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=json,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=lazy,Value=1,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=conf-reload-interval,Value=0,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=no-ssl-check,Value=1,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=user,,)
MSI (s) (E4:0C) [09:48:42:944]: Executing op: RegAddValue(Name=password,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=proxy,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=tasks,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=ca-cert-dir,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=ca-cert-file,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=ssl-cert-file,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=ssl-fingerprint,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=vardir,Value=C:\Program Files\GLPI-Agent\var,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=listen,Value=0,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=remote,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=remote-workers,Value=1,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegOpenKey(Root=-2147483646,Key=Software\GLPI-Agent\Installer,,BinaryType=1,,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=InstallDir,Value=C:\Program Files\GLPI-Agent\,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=ExecMode,Value=1,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=QuickInstall,Value=1,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=AddFirewallException,Value=1,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=RunNow,Value=1,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=TaskFrequency,Value=hourly,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=TaskMinuteModifier,Value=15,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=TaskHourlyModifier,Value=1,)
MSI (s) (E4:0C) [09:48:42:960]: Executing op: RegAddValue(Name=TaskDailyModifier,Value=1,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: RegAddValue(Name=AgentMonitor,Value=0,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: RegAddValue(Name=Version,Value=1.9,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: RegOpenKey(Root=-2147483646,Key=Software\GLPI-Agent\Monitor,,BinaryType=1,,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: RegAddValue(Name=NewTicket-URL,,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: ActionStart(Name=InstallServices,Description=Installing new services,Template=Service: [2])
MSI (s) (E4:0C) [09:48:42:976]: Executing op: ProgressTotal(Total=1,Type=1,ByteEquivalent=1300000)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: ServiceInstall(Name=glpi-agent,DisplayName=GLPI Agent,ImagePath="C:\Program Files\GLPI-Agent\perl\bin\glpi-agent.exe" -I"C:\Program Files\GLPI-Agent\perl\agent" -I"C:\Program Files\GLPI-Agent\perl\site\lib" -I"C:\Program Files\GLPI-Agent\perl\vendor\lib" -I"C:\Program Files\GLPI-Agent\perl\lib" "C:\Program Files\GLPI-Agent\perl\bin\glpi-win32-service",ServiceType=16,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=GLPI Agent is an inventory agent. It is intended to upload system inventory toward a GLPI server on a regular basis.,,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: ActionStart(Name=RollbackServiceConfig,,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: CustomActionSchedule(Action=RollbackServiceConfig,ActionType=3329,Source=BinaryData,Target=RollbackServiceConfig,CustomActionData=SchedServiceConfig)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: ActionStart(Name=ExecServiceConfig,,)
MSI (s) (E4:0C) [09:48:42:976]: Executing op: CustomActionSchedule(Action=ExecServiceConfig,ActionType=3073,Source=BinaryData,Target=ExecServiceConfig,CustomActionData=SchedServiceConfigglpi-agent1restartrestartrestart060)
MSI (s) (E4:0C) [09:48:42:976]: Creating MSIHANDLE (86) of type 790536 for thread 7436
MSI (s) (E4:9C) [09:48:42:976]: Invoking remote custom action. DLL: C:\windows\Installer\MSIBA36.tmp, Entrypoint: ExecServiceConfig
MSI (s) (E4!88) [09:48:42:991]: Creating MSIHANDLE (87) of type 790531 for thread 5256
MSI (s) (E4!88) [09:48:42:991]: Closing MSIHANDLE (87) of type 790531 for thread 5256
MSI (s) (E4:9C) [09:48:42:991]: Closing MSIHANDLE (86) of type 790536 for thread 7436
MSI (s) (E4:0C) [09:48:42:991]: Executing op: ActionStart(Name=ForceRun,,)
MSI (s) (E4:0C) [09:48:42:991]: Executing op: CustomActionSchedule(Action=ForceRun,ActionType=3137,Source=BinaryData,Target=WixQuietExec,CustomActionData="C:\Program Files\GLPI-Agent\glpi-agent.bat" --set-forcerun)
MSI (s) (E4:0C) [09:48:42:991]: Creating MSIHANDLE (88) of type 790536 for thread 7436
MSI (s) (E4:D8) [09:48:42:991]: Invoking remote custom action. DLL: C:\windows\Installer\MSIBA46.tmp, Entrypoint: WixQuietExec
MSI (s) (E4:D8) [09:53:55:478]: Closing MSIHANDLE (88) of type 790536 for thread 7436
MSI (s) (E4:0C) [09:53:55:478]: Executing op: ActionStart(Name=StartServices,Description=Starting services,Template=Service: [1])
MSI (s) (E4:0C) [09:53:55:478]: Executing op: ProgressTotal(Total=1,Type=1,ByteEquivalent=1300000)
MSI (s) (E4:0C) [09:53:55:478]: Executing op: ServiceControl(,Name=glpi-agent,Action=1,Wait=1,)
MSI (s) (E4:0C) [09:57:54:712]: Product: GLPI Agent 1.9 -- Error 1920. Service 'GLPI Agent' (glpi-agent) failed to start. Verify that you have sufficient privileges to start system services.

Error 1920. Service 'GLPI Agent' (glpi-agent) failed to start. Verify that you have sufficient privileges to start system services.
```

**Comment by g-bougard** [comment]
> I wont use official vbs script, because I don't have internet connection on this PCs.

You can use vbs script to install a local file or from a shared storage.

> `Error 1920. Service 'GLPI Agent' (glpi-agent) failed to start. Verify that you have sufficient privileges to start system services.`

Can you confirm you're attempting installation with sufficient privileges ?

This is definitively not related to #584 as here the installer fails to install service.

Last point, do you have GLPI-Agent installed when running installer ? And if yes, is it installed in x86/32 bits version ? If this is the case, you must uninstall first the 32 bits version before attempting to install de 64 bits one.

**Comment by danix1** [comment]
Thank you for your response.

Now I tried to use vbs script with network shared storage.
And same problem, version 1.7.3 installing with vbs script is working OK. When I try to install 1.8 with vbs, 10 minutes stuck and not installed. Same storage, same permission, same parameters.

I always use 64bit version, and I also try install with previous version installed 1.7.3, and also without any installed version of glpi-agent. I also try to manually install with GUI installer, and same problem, in the end installer is trying start service, and waiting minutes with error. Version 1.7.3 with GUI installed and working OK.
Computer is new with clean Windows 11 Pro 23H2, with last cummulative 2024-05 update.

**Reply by g-bougard** [reply]
Do you have any active antivirus ? And if yes, can you check if it reports an action during glpi-agent installation ?

Can you share the `SetupOptions` variable you set in vbs ? You can of course obfuscate the SERVER value.
It seems the installer fails on starting the service so maybe a wrong configuration prevents it to start.

Check eventually system EventViewer for other related errors.

**Reply by danix1** [reply]
I couldn't find any report in antivirus - Eset endpoint antivirus 11.

My variable in vbs is: SetupOptions = "/quiet RUNNOW=1 ADD_FIREWALL_EXCEPTION=0 QUICKINSTALL=1 SERVER='https://glpi.domain.com/plugins/glpiinventory'"

In EventViewer I only see same error: Service 'GLPI Agent' (glpi-agent) failed to start. Verify that you have sufficient privileges to start system services.

Same settings for version 1.7.3, which is installed and running ok, but newest version 1.8 and 1.9 failed to install. I try also nightly build 1.10 but same problem.

On Windows 10 systems (with last cummulative 2024-05 update) this new versions 1.8 and 1.9 works OK, same antivirus and configuration.

**Reply by g-bougard** [reply]
QUICKINSTALL=1 is not required, this is only an option for interactive installation. Have you tried with ADD_FIREWALL_EXCEPTION=1 ?

Anyway, as far I understand, finally the problem is only on Windows 11, right ?

**Reply by danix1** [reply]
Hi, Yes problem is only on Windows 11 Pro, PC is domain joined. I try another combination of install parameters, with ADD_FIREWALL_EXCEPTION=1, RUNNOW=1, NO_SSL_CHECK=1 etc. Nothing is working with version 1.9. Try to completely uninstall antivirus, disable firewall. Previous version of glpi-agent service is not present on system, I try show service status with: sc query glpi-agent.

I ended with GUI install, try to install without inventory to glpi server, without internet connection, with Local target: c:\Program files\GLPI-agent and problem is same, service not started error. When I try same settings to install version 1.7.3 instantly working, fast install, service is working, inventory working too.

Maybe problem is in some glpi-agent changes, perl or library changes from version 1.7.3. I must stay on this old version, I can't find any solution for install version 1.8 or 1.9 as a windows system service. I can try any debug or testing, if you send me information, how to run that. Thank you.

**Reply by g-bougard** [reply]
Hi @danix1 

maybe there's a bug in the Windows 11 Pro OS which will be fixed at some time in the future. You may eventually contact Microsoft support to know if there's a known issue with some other software installed as a service.

Maybe this is a WixToolSet issue (we uses actually v3.14.1) as the service installation is started by a WixToolSet CustomAction. A future upgrade can help in that case, but it is not planned right now.

You still provided a MSI debug log, so you still did a good debug of the problem. I don't see actually how to debug more deeply.

If you want to go deeper, maybe you can first install the agent in manual mode. Then try to run manually the agent to see first if it just works this way. Not running this way can explain why the installer fails to start the service and decides to abort installation.

After that if the agent works manually, you can try to install the service with such command run from an administrative console and from GLPI-Agent installation folder:
```
perl\bin\glpi-agent.exe perl\bin\glpi-win32-service --register -n glpi-agent -d "GLPI Agent"
```
This is another way to register the service I used sometime in windows development environment. There's an online manual page here: https://glpi-agent.readthedocs.io/en/latest/man/glpi-win32-service.html. Here in my proposed commandline, `perl\bin\glpi-agent.exe` is really the installed perl.exe on windows.

If you're able to run the agent manually and install the service this way, we will be sure the issue is around MSI service installation.

**Reply by danix1** [reply, marked as answer]
I find a solution. Windows 11 added Smart App Control to Windows Defender antivirus. Default setting is Evaluation mode. After setting Smart App Control to off, everything is working correctly now.
How to disable Smart App Control - https://www.elevenforum.com/t/turn-on-or-off-smart-app-control-in-windows-11.4996

I tested latest glpi-agent 1.10, and everything is working correctly now. Installation with GPO startup script, service is running and inventory is working correctly.
Glpi-agent won't run even on offline mode - zip portable version, with disabled Smart App Control, it run correctly again. Thank you.

**Reply by danix1** [reply]
I updated a solution
